lib/match.rb

```diff
@@ -0,0 +1,98 @@
+require_relative "board"
+require_relative "player"
+require_relative 'validation.rb' 
+require_relative 'turn_management.rb' 
+
+class Match
+
+    include Validation
+    include Turns 
+
+    def initialize(board, player_1, player_2)
+
+        @board = board 
+        @player_1 = player_1
+        @player_2 = player_2
+        @turn_type = setup_turns
+        @winner_type = ""
+    end
+
+    def game_loop
+
+        @board.welcome_message(@player_1.token, @player_2.token)
+
+        until game_over?
+
+            @turn_type = set_next_turn(@turn_type)
+            user_move
+            game_state
+
+        end 
+
+        decide_winner_or_tie
+
+    end 
+
+    # Method to evaluate a winner, a tie,
+    private
+
+    def game_state
+        @board.display
+    end 
+
+
+    def game_over?
+        winner? || tie?
+    end 
+     
+
+    def winner?
```

Comment:
Hi guys. A suggested change here is to split this `winner?` method into 3 -> a method that checks for vertical, one for horizontal and one for diagonal.. Then `winner?` will just need to call those 3.